rpc_util.go

```diff
@@ -867,19 +869,43 @@ func recvAndDecompress(p *parser, s recvCompressor, dc Decompressor, maxReceiveM
 }
 
 // Using compressor, decompress d, returning data and size.
-// Optionally, if data will be over maxReceiveMessageSize, just return the size.
-func decompress(compressor encoding.Compressor, d mem.BufferSlice, maxReceiveMessageSize int, pool mem.BufferPool) (mem.BufferSlice, int, error) {
+// If the decompressed data exceeds maxReceiveMessageSize, it returns errMaxMessageSizeExceeded.
+func decompress(compressor encoding.Compressor, d mem.BufferSlice, maxReceiveMessageSize int, pool mem.BufferPool) (mem.BufferSlice, error) {
 	dcReader, err := compressor.Decompress(d.Reader())
 	if err != nil {
-		return nil, 0, err
+		return nil, err
 	}
 
-	out, err := mem.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)+1), pool)
+	out, err := mem.ReadAll(io.LimitReader(dcReader, int64(maxReceiveMessageSize)), pool)
 	if err != nil {
 		out.Free()
-		return nil, 0, err
+		return nil, err
+	}
+
+	if doesReceiveMessageOverflow(int64(out.Len()), int64(maxReceiveMessageSize), dcReader) {
```

Comment:
Why are these values being cast into int64?

Reply:
avoiding Overflow like If out.Len() or maxReceiveMessageSize are returned as smaller numeric types (like int32 or int on a 32-bit system), casting them to int64 ensures that the arithmetic or comparison doesn’t risk overflow

Reply:
There doesn't seem to be any addition done in `doesReceiveMessageOverflow()`, so where might the overflows happen?

Reply:
You're correct that the doesReceiveMessageOverflow function only performs a comparison and does not involve any arithmetic operations.

Reply:
I modified the code and push it.

Reply:
Please let reviewers resolve the conversations.. Thank you.